app/bundles/EmailBundle/Model/EmailModel.php

```diff
@@ -724,6 +729,32 @@ public function getEmailStati($emailId, $leadId)
         );
     }
 
+    /**
+     * @param Email $entity
+     *
+     * @return array<string, array<int, array<string, int|string>>>
+     *
+     * @throws \Doctrine\DBAL\Exception
+     */
+    public function getCountryStats($entity, \DateTime $dateFrom, \DateTime $dateTo, bool $includeVariants = false): array
```

Comment:
Use DateTimeInterface wherever possible. We should move to DateTimeImmutable objects in the future and using the interface will make it easier.